src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs

```diff
@@ -11,113 +11,342 @@
+                _movementThreshold = value;
+                RemoveUpdates();
+                RequestUpdates();
+            }
+        }
+
+        public uint? DesiredAccuracyInMeters
+        {
+            get => _desiredAccuracyInMeters;
+            set
+            {
+                _desiredAccuracyInMeters = value;
+                RemoveUpdates();
+                RequestUpdates();
+            }
+        }
+
+        #region MyListener
+
+        public class MyLocListen : Java.Lang.Object, ILocationListener
+        {
 
-		public Task<Geoposition> GetGeopositionAsync(TimeSpan maximumAge, TimeSpan timeout)
-			=> GetGeopositionAsync();
+            public void OnLocationChanged(Location location)
+            {
+                DateTime date = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+                date = date.AddMilliseconds(location.Time);
+                if (date.AddMinutes(1) > DateTime.UtcNow)
+                { // only from last minute (we don't want to get some obsolete location)
+                    mbZmiany = true;
+                    moLocat = location;
+                }
+            }
 
-		public static async Task<GeolocationAccessStatus> RequestAccessAsync() => GeolocationAccessStatus.Allowed;
+            public void OnProviderDisabled(string provider)
+            {
+                throw new global::System.OperationCanceledException();
+            }
 
-		private LocationManager InitializeLocationProvider(double desiredAccuracy)
-		{
-			var locationManager = (LocationManager)global::Android.App.Application.Context.GetSystemService(Android.Content.Context.LocationService);
+            public void OnProviderEnabled(string provider)
+            {
+                
+            }
 
-			var criteriaForLocationService = new Criteria
-			{
-				Accuracy = Accuracy.Coarse
-			};
+            public void OnStatusChanged(string provider, Availability status, Bundle extras)
+            {
+                // This method was deprecated in API level 29 (Android 10). This callback will never be invoked.
+            }
+        }
 
-			var acceptableLocationProviders = locationManager.GetProviders(criteriaForLocationService, true);
 
-			if (acceptableLocationProviders.Any())
-			{
-				_locationProvider = acceptableLocationProviders.First();
-			}
-			else
-			{
-				_locationProvider = String.Empty;
-			}
+        MyLocListen mojSluchacz = new MyLocListen();
+        #endregion 
 
-			return locationManager;
-		}
+        public Geolocator()
+        {
+            _locationManager = (LocationManager)Android.App.Application.Context.GetSystemService(Android.Content.Context.LocationService);
 
-		partial void StartPositionChanged()
-		{
+            _reportInterval = 1000;
+            _movementThreshold = 0;
+
+            RequestUpdates();
+        }
+
+        ~Geolocator()
+        {
+            try
+            {
+                RemoveUpdates();
+            }
+            catch
+            { }
```

Comment:
What are the expected exceptions here? Why are they silenced?

Reply:
RemoveUpdates called twice

Reply:
That exception should be handled in the `RemoveUpdates()` directly and it should be to the known exception.

If you know a way to prevent an exception from throwing it would be even better.